teammapper-backend/src/map/services/maps.service.ts

```diff
@@ -62,27 +62,24 @@ export class MapsService {
     )
   }
 
-  async addNode(mapId: string, node: MmpNode): Promise<MmpNode> {
+  async addNode(mapId: string, node: MmpNode): Promise<MmpNode | null> {
```

Comment:
Just a quick note: What we always have to check is, if the parent actually exists. Upsert isn't solving this. It's the same for add, update and delete. If the parent was deleted before, it makes no sense to proceed.

Reply:
on the other hand, we are already catching errors further below and we can't completely prevent the risk with a check. so as long as we don't have sophisticated syncing mechanisms in place, we maybe can just fallback on that.